specification/network/resource-manager/Microsoft.Network/stable/2018-02-01/expressRouteCrossConnection.json

```diff
@@ -0,0 +1,738 @@
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The path of the device."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns the resulting ExpressRouteCrossConnectionsRouteTableSummary resource.",
+            "schema": {
+            "$ref": "./expressRouteCircuit.json#/definitions/ExpressRouteCircuitsRoutesTableSummaryListResult"
+            }
+          },
+          "202": {
+            "description": "Accepted and the operation will complete asynchronously."
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/expressRouteCrossConnections/{crossConnectionName}/peerings/{peeringName}/routeTables/{devicePath}": {
+      "post": {
+        "tags": [
+          "ExpressRouteCrossConnectionRouteTable"
+        ],
+        "operationId": "ExpressRouteCrossConnections_ListRoutesTable",
+        "description": "Gets the currently advertised routes table associated with the express route crossConnection in a resource group.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "crossConnectionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the ExpressRouteCrossConnection."
+          },
+          {
+            "name": "peeringName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the peering."
+          },
+          {
+            "name": "devicePath",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The path of the device."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+              "description": "Request successful. The operation returns the resulting ExpressRouteCrossConnectionsRouteTable resource.",
+              "schema": {
+                "$ref": "./expressRouteCircuit.json#/definitions/ExpressRouteCircuitsRoutesTableListResult"
+              }
+          },
+          "202": {
+            "description": "Accepted and the operation will complete asynchronously."
+          }
+        },
+        "x-ms-long-running-operation": true
```

Comment:
Is there a reason why this list operation is long running

Reply:
The Cross Connection resource shadows the corresponding Express Route Circuit resource. The list route table operation would fetch the contents for the route tables from the device for multiple tables. Further each table may have several records.